lib/ruby_smb/smb2/packet/session_setup_request.rb

```diff
@@ -30,6 +30,17 @@ def set_type1_blob(type1_message)
           self.buffer = gss_blob
         end
 
+        # Takes a serialized NTLM Type 1 message inserts it into the {RubySMB::SMB2::Packet::SessionSetupRequest#buffer}
+        # as well as updating the {RubySMB::SMB2::Packet::SessionSetupRequest#security_buffer_length}.
+        # This method DOES NOT wrap the NTLM Type 1 message in GSS.
+        #
+        # @param type1_message [String] the serialized NTLM Type 1 message
+        # @return [void]
+        def set_ntlm_type1_blob(type1_message)
+          self.security_buffer_length = type1_message.length
+          self.buffer = type1_message
+        end
```

Comment:
We'll probably want a better pattern here. Related thought, a potential rename of the newly added `gss_provider` namespace to something more generic may be beneficial as well